### PR TITLE
Added "Open in: new tab, browser player, streamlink" in twitch link context menu.

### PR DESCRIPTION
**Pull request by SLCH**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Added "Open in: new tab, browser player, streamlink" in twitch link context menu.

<details>
<summary>
New - Old
</summary>

New:

Screenshot_20210711_120131

Old:

Screenshot_20210711_120050

</details>

### CODE REVIEW

**Comment by SLCH** [comment]
```
Error: fatal: No url found for submodule path 'lib/humanize' in .gitmodules
```
Getting in actions on submodule loading.
This lib was removed around January.
Not sure why action is trying to load it - there is no mention of it in my diff.

**Comment by ALazyMeme** [comment]
Could be because you added the library?
https://github.com/SLCH/chatterino2/commit/a947282522e524693e61b9a57e7bcfdf0baa4826

File: resources/contributors.txt [hunk 1]
```diff
@@ -42,6 +42,7 @@ Iulian Onofrei | https://github.com/revolter | :/avatars/revolter.jpg | Contribu
 matthewde | https://github.com/m4tthewde | :/avatars/matthewde.jpg | Contributor
 Karar Al-Remahy | https://github.com/KararTY | :/avatars/kararty.png | Contributor
 Talen | https://github.com/talneoran | | Contributor
+SLCH | https://github.com/SLCH | | Contributor
```
Comment (ALazyMeme) [inline resources/contributors.txt:45]: Isn't this meant to be in a separate pr?

Comment (SLCH) [reply resources/contributors.txt:45]: i have no idea, could reverse it.

**Comment by SLCH** [comment]
> Could be because you added the library?
> [SLCH@a947282](https://github.com/SLCH/chatterino2/commit/a947282522e524693e61b9a57e7bcfdf0baa4826)

I did not add it. It was removed since my last PR but i did update my master before writing any changes.
And i checked my diff with current master's and there is no mention of said lib.

File: resources/contributors.txt (hunk 1 above)
Comment (SLCH) [reply resources/contributors.txt]: Reverted

Comment (Felanbird) [reply resources/contributors.txt]: > i have no idea, could reverse it.

Better to be a separate PR incase reverting of this PR happens, so your Contributor entry isn't lost.

Comment (zneix) [reply resources/contributors.txt]: It can be done in the same PR - I did this myself back in the old days. Feel free to include it in this one, I think there's no harm but if you wanna farm some more commits do it separetly.

Comment (zneix) [reply resources/contributors.txt]: > incase reverting of this PR happens

Why would it though? :thinking:

File: src/widgets/splits/Split.hpp [hunk 2]
```diff
@@ -114,6 +114,10 @@ class Split : public BaseWidget, pajlada::Signals::SignalHolder
     void handleModifiers(Qt::KeyboardModifiers modifiers);
     void updateInputPlaceholder();
 
+    void openChannelInBrowserPlayer(ChannelPtr channel);
+    void openChannelInStreamlink(QString channelName);
+    void joinChannelInNewTab(ChannelPtr channel);
```
Comment (pajlada) [inline src/widgets/splits/Split.hpp:119]: Could you add brief comments here explaining what the functions do?

File: src/widgets/splits/Split.cpp [hunk 3]
```diff
@@ -163,9 +163,30 @@ Split::Split(QWidget *parent)
...
+            case (FromTwitchLinkOpenChannelIn::Split):
+                this->openSplitRequested.invoke(channel);
+                break;
+            case (FromTwitchLinkOpenChannelIn::Tab):
+                this->joinChannelInNewTab(channel);
+                break;
+            case (FromTwitchLinkOpenChannelIn::BrowserPlayer):
+                this->openChannelInBrowserPlayer(channel);
+                break;
+            case (FromTwitchLinkOpenChannelIn::Streamlink):
```
Comment (pajlada) [inline src/widgets/splits/Split.cpp:182]: please remove the parenthesis from the cases

File: src/widgets/splits/Split.cpp [hunk 4]
```diff
@@ -309,6 +330,41 @@ void Split::updateInputPlaceholder()
     this->input_->ui_.textEdit->setPlaceholderText(placeholderText);
 }
 
+void Split::joinChannelInNewTab(ChannelPtr channel)
+{
+    // Code opening a new tab was taken from:
+    // src/widgets/dialogs/switcher/NewTabItem.cpp#22 - NewTabItem::action
```
Comment (pajlada) [inline src/widgets/splits/Split.cpp:336]: Useful comment for me now while reviewing it, but unnecessary to keep in the code

**Review by pajlada — CHANGES_REQUESTED** [review]
Small nitpicks, otherwise seems good :ok_hand:

**Review by pajlada — APPROVED** [review]

File: src/widgets/splits/Split.cpp [hunk 5]
```diff
@@ -595,13 +649,7 @@ void Split::openWhispersInBrowser()
...
 void Split::openBrowserPlayer()
 {
-    ChannelPtr channel = this->getChannel();
-    if (auto twitchChannel = dynamic_cast<TwitchChannel *>(channel.get()))
-    {
-        QDesktopServices::openUrl(
-            "https://player.twitch.tv/?parent=twitch.tv&channel=" +
-            twitchChannel->getName());
-    }
+    this->openChannelInBrowserPlayer(this->getChannel());
```
Comment (zneix) [inline src/widgets/splits/Split.cpp:652]: We could try to just replace (and potentially delete) `Split::openBrowserPlayer` with `Split::openChannelInBrowserPlayer` since we just have 2 functions that do literally the same, but maybe that's something that should be done in separate PR so just making a note for the future.

**Review by zneix — APPROVED** [review]
Looks good, just got one thought to share - potentially for the future PR though.

File: src/widgets/splits/Split.cpp (hunk 5 above)
Comment (SLCH) [reply src/widgets/splits/Split.cpp:652]: Duplicated cause `Split::openBrowserPlayer` was a qt slot and meant for current Split while `Split::openChannelInBrowserPlayer` takes any channel. Same about `Split::openChannelInStreamlink`.

Comment (zneix) [reply src/widgets/splits/Split.cpp:652]: > Duplicated cause Split::openBrowserPlayer was a qt slot and meant for current Split

Correct! Passing an empty channel or simply the current split's channel could be the way to go imho.

Comment (SLCH) [reply src/widgets/splits/Split.cpp:652]: > Correct! Passing an empty channel or simply the current split's channel could be the way to go imho.

I implemented it like this at first (with default channel arg = nulptr) but it wouldn't compile and later changed to current version. 
Something to do with slots. My C++ isn't very good :smile:
